extension/encoding/avrologencodingextension/extension.go

```diff
@@ -16,26 +16,51 @@ import (
 )
 
 var (
+	_ encoding.LogsMarshalerExtension   = (*avroLogExtension)(nil)
 	_ encoding.LogsUnmarshalerExtension = (*avroLogExtension)(nil)
 )
 
 type avroLogExtension struct {
-	deserializer avroDeserializer
+	config  *Config
+	encoder avroSerDe
 }
 
 func newExtension(config *Config) (*avroLogExtension, error) {
-	deserializer, err := newAVROStaticSchemaDeserializer(config.Schema)
+	encoder, err := newAVROStaticSchemaSerDe(config.Schema)
 	if err != nil {
 		return nil, err
 	}
 
-	return &avroLogExtension{deserializer: deserializer}, nil
+	return &avroLogExtension{
+		config:  config,
+		encoder: encoder,
+	}, nil
+}
+
+func (e *avroLogExtension) MarshalLogs(logs plog.Logs) ([]byte, error) {
+	body := logs.ResourceLogs().At(0).ScopeLogs().At(0).LogRecords().At(0).Body()
```

Comment:
what if there are no logs, or multiple logs, or multiple scope logs?

Reply:
Same implementation as in the `jsonlogencoding`. Reason for that, it is not possible to apply any generic logic that will handle multiple logs or scope logs and transform them into the provided schema. So it has to be one input one output.

Reply:
Well, sure, but could you possibly marshal them all individually and send back the data as separate payloads? You are going to force customers to have one log record per payload which is a bit of a trap. The same feedback applies to `jsonlogencoding`, looks like. I will open an issue for that.

Reply:
@atoulme yes, this needs to be addressed for `jsonlogencoding`. I open up a PR with the fix unless you are planning to do so. 👍

Reply:
Not against doing so, can you elaborate on how that could be implemented? `MarshalLogs` does only return `[]byte`.

Reply:
no clue, would maybe your schema account for multiple log records? Or is avro able to have a separator of some sort between records?

Reply:
@thmshmm I think you can make use of https://pkg.go.dev/encoding/json#Encoder.Encode. 
Something like following in go syntax:
```
func MarshalLogs(logs plog.Logs) ([]byte, error) {
    b = new(bytes.Buffer)
    encoder := json.NewEncoder(b)
    for record in logs:
       encoder.Encode(record.AsMap())
    return b.Bytes()
}
```

Reply:
You can also set any set any using `SetIndent`.

Reply:
We have to make sure not to break any of the consumers of the data. E.g. sending it to Kafka one message is expected to contain one Avro record. To keep the solution generic, without a fully predefined schema, we could try to define a root Avro record and merge it with the user provided schema. So the provided record schema becomes an array field in the root element. Or go with the restriction of only emitting one log but keeping full flexibility on the schema.

Reply:
> @atoulme yes, this needs to be addressed for `jsonlogencoding`. I'll open up a PR with the fix unless you are planning to do so. 👍

I have filed https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34064 for this, but I don't intend to bring a PR for it. All yours.